packages/vite/src/node/build.ts

```diff
@@ -401,17 +401,14 @@ export function resolveBuildEnvironmentOptions(
 
   // @ts-expect-error Fallback options instead of merging
   const resolved: ResolvedBuildEnvironmentOptions = {
-    target: 'modules',
-    cssTarget: false,
+    ...configDefaults.build,
     ...userBuildEnvironmentOptions,
     commonjsOptions: {
-      include: [/node_modules/],
-      extensions: ['.js', '.cjs'],
+      ...configDefaults.build.commonjsOptions,
       ...userBuildEnvironmentOptions.commonjsOptions,
     },
     dynamicImportVarsOptions: {
-      warnOnError: true,
-      exclude: [/node_modules/],
+      ...configDefaults.build.dynamicImportVarsOptions,
       ...userBuildEnvironmentOptions.dynamicImportVarsOptions,
     },
```

Comment:
Probably needs a helper function that merges recursively but replaces arrays, slightly different from `mergeConfig`.

Reply:
Added `mergeWithDefaults` function 👍